l10n_ch_scan_bvr/wizard/scan_bvr.py

```diff
@@ -485,61 +348,64 @@ def validate_bvr_string(self, cr, uid, ids, context):
         # Explode and check  the BVR Number and structurate it
         #
         data = {}
-        data['bvr_struct'] = self._get_bvr_structurated(
-            bvr_data.bvr_string)
+        data['bvr_struct'] = self._get_bvr_structurated(self.bvr_string)
+        partner_bank_model = self.env['res.partner.bank']
+        partner_bank = False
         # We will now search the account linked with this BVR
         if data['bvr_struct']['domain'] == 'name':
             domain = [('acc_number', '=', data['bvr_struct']['beneficiaire'])]
-            partner_bank_search = self.pool.get('res.partner.bank').search(
-                cr,
-                uid,
-                domain,
-                context=context
-            )
+            partners_bank = partner_bank_model.search(domain)
+            partner_bank = len(partners_bank) == 1 and partners_bank[0]
         else:
-            domain = [
-                ('bvr_adherent_num', '=', data['bvr_struct']['bvrnumber'])
-            ]
-            partner_bank_search = self.pool.get('res.partner.bank').search(
-                cr,
-                uid,
-                domain,
-                context=context
-            )
+            # A postal account that refers to a bank is uniquely identified
+            # by (beneficiaire + bvrnumber), at least by beneficiaire
+            domain = [('acc_number', '=', data['bvr_struct']['beneficiaire'])]
```

Comment:
Could we replace part of your code simply by searching on both `acc_number` and `bvr_reference` at the same time?

``` python
domain = [('acc_number', '=', data['bvr_struct']['beneficiaire']),
          ('bvr_adherent_num', '=', data['bvr_struct']['bvrnumber'])]
```

Be also aware that you have postal number in `ccp` field and not in `acc_number` anymore.
This is because you can set the IBAN number for your postal account (in the future IBAN will most likely replace ccp).